aten/src/ATen/OpaqueTensorImpl.h

```diff
@@ -99,6 +100,7 @@ c10::intrusive_ptr<TensorImpl> shallow_copy_and_detach() const override {
   impl->is_contiguous_ = is_contiguous_;
   impl->is_wrapped_number_ = is_wrapped_number_;
   impl->reserved_ = reserved_;
+  impl->memory_format_tag_ = memory_format_tag_;
```

Comment:
@gchanan Do you know why the field copy is implemented here? I would have expected us to defer to TensorImpl parent implementation here.

Reply:
It's really irritating that we needed to update *every* subclass of `TensorImpl` when we made this change. And what if we missed some? Ugh!

Reply:
I believe the latest version of https://github.com/pytorch/pytorch/pull/17072 rectifies this -- CC @yf225.

Reply:
The latest version of #17072 has `copy_tensor_data(src_impl, dest_impl)` function in TensorImpl which handles the copying of all base TensorImpl fields.